### PR TITLE
add new interface option for skipping interfaces

### PR DESCRIPTION
**Pull request by mgleung**
## Description
Added an option to skip network interfaces specified by regexes so that users can have first-found logic without having to manually specify system/setup specific interfaces to skip by changing the code.

Is dependent upon https://github.com/projectcalico/calico/pull/878 to function properly

Addresses #1555 

## Testing
Ran this against an image from https://github.com/projectcalico/calico/pull/878 and verified that the specified interfaces were skipped if they matched the regexes given

### CODE REVIEW

**Comment by bcreane** [comment]
lgtm!

File: calicoctl/commands/node/run.go [hunk 1]
```diff
@@ -531,6 +538,19 @@ func validateIpAutodetectionMethod(method string, version int) {
 			os.Exit(1)
 		}
 		return
+	} else if strings.HasPrefix(method, AUTODETECTION_METHOD_SKIP_INTERFACE) {
+		// Auto-detection method is "skip-interface", validate that the
+		// interface regexes used are valid golang regexes.
+		ifStr := strings.TrimPrefix(method, AUTODETECTION_METHOD_SKIP_INTERFACE)
+		// regexes are provided in a string separated by ","
```
Comment (caseydavenport) [inline calicoctl/commands/node/run.go]: nit: Should have a newline before this comment.
nit: Should capitalize the beginning of a comment.

**Comment by bcreane** [comment]
Hey Matt, suggest you squash the commits, probably down to 1, to make it easy to revert if we need to.

File: glide.lock [hunk 2]
```diff
@@ -1,5 +1,5 @@
 hash: 91a8d9a89a8e34c483b837ccd059cb40ebcea88394a9e6d9360425a749a0c7f2
```
Comment (bcreane) [inline glide.lock]: is glide.lock change intended?

Comment (mgleung) [inline glide.lock]: Yeah, its not intended, I was testing some other things and this got updated.  I will revert.

File: calicoctl/commands/node/run.go [hunk 3]
```diff
@@ -526,9 +535,28 @@ func validateIpAutodetectionMethod(method string, version int) {
...
+	} else if strings.HasPrefix(method, AUTODETECTION_METHOD_SKIP_INTERFACE) {
+		// Auto-detection method is "skip-interface", validate that the
+		// interface regexes used are valid golang regexes.
+		ifStr := strings.TrimPrefix(method, AUTODETECTION_METHOD_SKIP_INTERFACE)
+
+		// Regexes are provided in a string separated by ","
+		ifRegexes := strings.Split(ifStr, ",")
+		for _, ifRegex := range ifRegexes {
+			if _, err := regexp.Compile(ifRegex); err != nil {
+				fmt.Printf("Error executing command: invalid interface skipping regex specified for IP autodetection: %s\n", ifRegex)
```
Comment (caseydavenport) [inline calicoctl/commands/node/run.go]: nit: can we reword to remove the word "skipping" from this?

I think `invalid interface regex` gets the point across well enough - as is it sounds like it's going to skip the invalid regex, when actually it's just exiting.

Comment (mgleung) [inline calicoctl/commands/node/run.go]: Sounds good.  I wanted originally to have something in there that was a little more specific as to what exactly was failing, but it does create more issues in plain English...
